src/main/java/com/fundynamic/d2tm/game/math/BoundedVector2D.java

```diff
@@ -0,0 +1,33 @@
+package com.fundynamic.d2tm.game.math;
+
+
+public class BoundedVector2D<T extends Number> extends Vector2D {
```

Comment:
Also, `BoundedVector2D` extends `Vector2D`, but doesn't this violate LSP? I.e. can we always safely use a BoundedVector2D when a Vector2D is expected?